src/backend/commands/trigger.c

```diff
@@ -1164,6 +1164,15 @@ CreateTriggerFiringOn(CreateTrigStmt *stmt, const char *queryString,
 	referenced.objectSubId = 0;
 	recordDependencyOn(&myself, &referenced, DEPENDENCY_NORMAL);
 
+	/*
+	 * For composite triggers, add dependency from trigger function
+	 * to trigger so that a drop trigger will result in cascade drop
+	 * for function as well. Trigger functions are created as part of
+	 * create trigger for composite triggers.
+	 */
+	if (is_composite_trigger)
+		recordDependencyOn(&referenced, &myself, DEPENDENCY_NORMAL);
```

Comment:
1. Is there a test case for this?
2. Why we need `DEPENDENCY_NORMAL` instead of `DEPENDENCY_INTERNAL`?

Reply:
yes, I will raise that as part of extension changes.
I am just rearranging this code from extension to here and it was NORMAL dependency always.